### PR TITLE
Invite Admin Sequence

### PR DESCRIPTION
**Pull request by weronica**
- Implemented admin invitation sequence. Registration sequence is almost complete. 
- Updated `requirements.txt` to no longer include `flask-mail`.
- Wrote helper functions to send emails. 
- Removed old/obsolete/broken code.
- Began added docstings to functions, models, and forms.

### CODE REVIEW

File: app/auth/forms.py [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+		if include_organization_name:
+			self.organization_name.data = user.organization_name
+
+		if include_role_in_organization:
+			self.role_in_organization.data = user.role_in_organization
+
+		if include_email:
+			self.email.data = user.email
+
+class RegisterAdminForm(InviteAdminForm):
```
Comment (jondubin) [inline app/auth/forms.py]: Question: what's the difference between RegisterAdminForm and InviteAdmin form?

File: app/auth/forms.py [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+from magic_numbers import *
+from flask import current_app
+from flask.ext.wtf import Form
+from wtforms.validators import InputRequired, Length, Email, Optional
+from wtforms import TextField, PasswordField
+from ..models import User, InvitedUser
+
+class InviteAdminForm(Form):
+	"""Form that is used to invite a new administrator."""
+	first_name = TextField(
```
Comment (jondubin) [inline app/auth/forms.py]: Nit: the TextField alias for StringField is now deprecated

File: app/auth/forms.py (hunk 1 above)
Comment (weronica) [inline app/auth/forms.py]: `InviteAdminForm` is used by a current administrator to invite a new admin, while `RegisterAdminForm` is used by the new admin to confirm their registration on the site. Do you have other names in mind that might be less ambiguous?

File: app/auth/forms.py (hunk 2 above)
Comment (weronica) [inline app/auth/forms.py]: Replaced. Also, it looks like `InputRequired` is now `DataRequired`. Also fixed.

Comment (jondubin) [inline app/auth/forms.py]: I read up on this last night, and there's a minor difference between the two: http://stackoverflow.com/questions/23982917/flask-wtforms-difference-between-datarequired-and-inputrequired 

I think we generally want InputRequired

File: app/auth/forms.py (hunk 1 above)
Comment (jondubin) [inline app/auth/forms.py]: Ahh, makes sense. I think these names work well.

Comment (weronica) [inline app/auth/forms.py]: Excellent.

File: app/auth/forms.py (hunk 2 above)
Comment (weronica) [inline app/auth/forms.py]: Ahh ok, got it. I added `InputRequired` back in.

**Pull request merged**